src/Pact/Types/PactError.hs

```diff
@@ -45,12 +47,43 @@ data StackFrame = StackFrame {
 instance NFData StackFrame
 instance ToJSON StackFrame where toJSON = toJSON . show
 
+-- | BIG HUGE CAVEAT: Back compat requires maintaining the pre-existing
+-- 'ToJSON' instance, so this is ONLY for UX coming out of serialized
+-- endpoints like `poll` in Chainweb; "Info" and "FunApp" values will
+-- be sketchy. As such this is also permissive on failure.
+instance FromJSON StackFrame where
+  parseJSON = withText "StackFrame" $ \t -> case parseOnly parseStackFrame t of
+    Right sf -> pure sf
+    Left e -> pure $ StackFrame ("StackFrame parse failed: " <> pack e) def Nothing
+
+
 instance Show StackFrame where
     show (StackFrame n i app) = renderInfo i ++ ": " ++ case app of
       Nothing -> unpack n
       Just (_,as) -> "(" ++ unpack n ++ concatMap (\a -> " " ++ unpack (asString a)) as ++ ")"
+
+-- | Attempt to parse 'Show' instance output. Intentionally avoids parsing app args,
+-- cramming all of the text into '_sfName'.
+parseStackFrame :: AP.Parser StackFrame
+parseStackFrame = do
+  i <- parseRenderedInfo
+  void $ string ": "
+  parseDeets i <|> justName i
+  where
+    parseDeets i = do
+      void $ char '('
+      deets <- T.init <$> takeText
```

Comment:
this will take up to `)` yeah?

Reply:
Yes, due to `init`. It doesn't actually enforce the last char is `)` though